### PR TITLE
Adds AccountId

### PR DESCRIPTION
**Pull request by 37ng**
# tl;dr

Adds `AccountId` struct
Adds `AccountId` <-> CAIP-10 String

Resolves this [comment](https://github.com/xmtp/libxmtp/pull/673#discussion_r1576869566)

### CODE REVIEW

File: xmtp_cryptography/src/signature.rs [hunk 1]
```diff
@@ -107,11 +108,6 @@ fn eip_191_prefix(msg: &str) -> String {
     format!("\x19Ethereum Signed Message:\n{}.", msg.len())
 }
 
-fn addr_string_to_bytes(str: &str) -> Result<Vec<u8>, SignatureError> {
-    let unprefixed_address = str::strip_prefix(str, "0x").unwrap_or(str);
-    hex::decode(unprefixed_address).map_err(SignatureError::BadAddressFormat)
-}
-
```
Comment (37ng) [inline xmtp_cryptography/src/signature.rs:114]: Removed this parsing logic and use ethers.rs function instead.

**Comment by neekolas** [comment]
Think this needs to be rebased on `main`

**Comment by 37ng** [comment]
> Think this needs to be rebased on `main`

Yep, done

File: xmtp_id/src/associations/serialization.rs [hunk 2]
```diff
@@ -413,4 +452,76 @@ mod tests {
 
         assert_eq!(serialized_update, reserialized);
     }
+
+    #[test]
+    fn test_accound_id() {
```
Comment (neekolas) [inline xmtp_id/src/associations/serialization.rs:457]: Great tests

**Review by neekolas — APPROVED** [review]

**Pull request merged**